libfreerdp/core/info.c

```diff
@@ -833,6 +835,9 @@ static void rdp_write_info_packet(rdpRdp* rdp, wStream* s)
 		                                0) * 2;
 	}
 
+	/* excludes (!) the length of the mandatory null terminator */
+	cbWorkingDir = cbWorkingDir > 0 ? cbWorkingDir - 2 : 0;
```

Comment:
I´d change that to `cbWorkingDir = cbWorkingDir > 2 ? cbWorkingDir - 2 : cbWorkingDir;` to avoid a possible overflow